backend/server.py

```
@@ -49,58 +52,134 @@ def process_input_text(input_text, use_mediawiki):
+def handle_unexpected_exception(e):
+    """Handle unexpected exceptions with a generic JSON response."""
+    return jsonify({
+        "error": "Internal server error",
+        "type": "InternalServerError",
+    }), 500
+
+
+def require_json_field(data, field_name):
+    """Validate JSON payload and return a required field."""
+    if not isinstance(data, Mapping):
+        raise BadRequest("JSON body must be an object")
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)